integration-cli/docker_cli_run_unix_test.go

```diff
@@ -824,9 +824,30 @@ func (s *DockerSuite) TestRunTmpfsMounts(c *check.C) {
 	if _, _, err := dockerCmdWithError("run", "--tmpfs", "/run:foobar", "busybox", "touch", "/run/somefile"); err == nil {
 		c.Fatalf("/run mounted on tmpfs when it should have vailed within invalid mount option")
 	}
-	if _, _, err := dockerCmdWithError("run", "--tmpfs", "/run", "-v", "/run:/run", "busybox", "touch", "/run/somefile"); err == nil {
```

Comment:
Sorry, just *1* more thing, I think the reason to have this is that the behavior would be undefined here and is most likely user error.
We should be able to keep this behavior if we catch it while iterating over binds in volumes.go.
Something like
```go
_, tmpfsExists := hostConfig.Tmpfs[bind.Destination]
if binds[bind.Destination] || tmpfsExists {
    return fmt.Errorf("Duplicate mount point '%s'", bind.Destination)
}
```

The reason `-v /test -v /tmp:/test` works is because of ambiguity in the API, something I hope to solve soon. You can see this in the services API where instead of `-v` we have `--mount type=bind,source=/tmp,target=/test --mount type=volume,target=/test`